From: Matt Fleming <matt.fleming@intel.com>
Date: Tue, 15 Nov 2011 12:57:03 +0000
Subject: [PATCH 05/11] efi.h: Add allocation types for
 boottime->allocate_pages()

commit bb05e4ba452ada7966fbced4e829aa029f546445 upstream.

Add the allocation types detailed in section 6.2 - "AllocatePages()"
of the UEFI 2.3 specification. These definitions will be used by the
x86 EFI boot stub which needs to allocate memory during boot.

Cc: Matthew Garrett <mjg@redhat.com>
Signed-off-by: Matt Fleming <matt.fleming@intel.com>
Link: http://lkml.kernel.org/r/1318848017-12301-1-git-send-email-matt@console-pimps.org
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
---
 include/linux/efi.h |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: linux-3.2.46/include/linux/efi.h
===================================================================
--- linux-3.2.46.orig/include/linux/efi.h	2013-07-26 19:19:45.000000000 +0000
+++ linux-3.2.46/include/linux/efi.h	2013-07-26 19:19:47.000000000 +0000
@@ -114,6 +114,14 @@
 	u32 imagesize;
 } efi_capsule_header_t;
 
+/*
+ * Allocation types for calls to boottime->allocate_pages.
+ */
+#define EFI_ALLOCATE_ANY_PAGES		0
+#define EFI_ALLOCATE_MAX_ADDRESS	1
+#define EFI_ALLOCATE_ADDRESS		2
+#define EFI_MAX_ALLOCATE_TYPE		3
+
 typedef int (*efi_freemem_callback_t) (u64 start, u64 end, void *arg);
 
 /*
